libs/model/src/test/seed.ts

```diff
@@ -0,0 +1,77 @@
+import { generateMock } from '@anatine/zod-mock';
+import { Model, ModelStatic } from 'sequelize';
+import z from 'zod';
+
+// props that are not mocked unless otherwise specified
+// via `allowedGeneratedProps`
+const GENERATED_PROPS = [
+  'id',
+  'created_at',
+  'updated_at',
+  'createdAt',
+  'updatedAt',
+  'deleted_at',
+] as const;
+type GeneratedProp = typeof GENERATED_PROPS[number];
+
+// incrementing seed number for each call to the `seed` function
+// for deterministic randomness when the order of calls is preserved
+let seedNum = 1;
+
+export type SchemaWithModel<T extends z.AnyZodObject> = {
+  schema: T;
+  model: ModelStatic<Model>;
+  mockDefaults?: () => Partial<z.infer<T>>;
+  allowedGeneratedProps?: GeneratedProp[];
+  buildQuery?: (data: z.infer<T>) => {
+    where: Partial<z.infer<T>>;
+  };
+};
+
+export type SeedOptions = {
+  mock: boolean;
+};
+
+export async function seed<T extends SchemaWithModel<any>>(
```

Comment:
Super small detail... a default `options: SeedOptions = { mock: true }` in `seed(...)` would make it explicit... and you can use `if(options.mock) {` later without having to define another variable.
I would also start using jsdoc comments in all exported functions to improve our documentation/intellisense